client/src/components/Input/SearchCategories.js

```diff
@@ -36,32 +39,42 @@ export default class SearchCategories extends React.Component {
     };
     this.renderTabsContainer = this.renderTabsContainer.bind(this);
     this.onSelectItem = this.onSelectItem.bind(this);
-    this.resetSelectedValue = this.resetSelectedValue.bind(this);
+    this.getCategoryFromQuery = this.getCategoryFromQuery.bind(this);
   }
 
-  resetSelectedValue() {
-    this.setState({ selectedValue: null });
+  componentDidMount() {
+    this.getCategoryFromQuery();
+    const unlisten = this.props.history.listen((location) => {
```

Comment:
does this listener need to be cleaned up?

Reply:
Yes, it is getting cleared within itself
```js
const unlisten = this.props.history.listen((location) => {
   if (location.pathname === "/feed") return this.getCategoryFromQuery();
   unlisten();
});
```